lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -345,6 +345,7 @@ namespace Js
 #endif
 
 #ifdef ENABLE_BASIC_TELEMETRY
+        // TODO - allocate this on the Heap instead of using a custom allocator?
         this->telemetry = Anew(this->TelemetryAllocator(), Js::ScriptContextTelemetry, this);
```

Comment:
Why not just inline the instance on the script context instead of allocating it?

Reply:
Yeah, IIRC, there were some weird deps in the headers that was preventing this, such that we need to use a forward declaration and pointer rather than the class.   I didn't get around to teasing the deps apart, which is why I left the //TODO.  I'm not clear on the impact of using a specific allocator here just for telemetry.  IIRC this allocator is only ever used for this specific allocation and nowhere else.

Reply:
Well, it just means that the instance is going to take up one page. 

Probably want to do it like SourceCodeAllocator and RegexAllocator, and redirect it to the generalAllocator normally, but separate under #ifdef SEPARATE_ARENA

---
In reply to: [187643452](https://github.com/Microsoft/ChakraCore/pull/5119#discussion_r187643452) [](ancestors = 187643452)